src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -4,31 +4,55 @@
  */
 
 import { createHash } from "crypto";
-
 import { K8s, kind } from "pepr";
 import { UDSConfig } from "../../../../config";
 import { Client } from "../types";
 import { buildChain, log } from "./authservice";
 import { Action, AuthserviceConfig } from "./types";
 
+let pendingSecretFetch: Promise<AuthserviceConfig> | null;
+
+// Cache for in-memory secret to avoid unnecessary Kubernetes secret lookups
+let inMemorySecret: AuthserviceConfig | null = null;
+
+// Track pending package updates and their resolve functions
+const pendingPackages: Map<
+  AuthserviceConfig,
+  { resolve: () => void; reject: (reason?: Error) => void }
+> = new Map();
+
+// Backup for the last known successful state of the secret
+let lastSuccessfulSecret: AuthserviceConfig | null = null;
+
+// Timer for debouncing updates to the secret
+let debounceTimer: NodeJS.Timeout | null = null;
+
+// Debounce duration (12 seconds) to reduce excessive updates, configurable via environment variable
+const DEBOUNCE_DURATION = parseInt(process.env.DEBONCE_DURATION || "1000", 10);
```

Comment:
Probably from my original code, typo here:

```suggestion
const DEBOUNCE_DURATION = parseInt(process.env.DEBOUNCE_DURATION || "1000", 10);
```

Reply:
Fixed